pkg/shipper/shipper.go

```diff
@@ -276,10 +279,16 @@ func (s *Shipper) Sync(ctx context.Context) (uploaded int, err error) {
 	)
 
 	uploadCompacted := s.uploadCompactedFunc()
-	metas, err := s.blockMetasFromOldest()
+	metas, failedBlocks, err := s.blockMetasFromOldest()
+	// Ignore error when there are failed blocks and we should ignore them
 	if err != nil {
-		return 0, err
+		if len(failedBlocks) == 0 || !s.skipCorruptedBlocks {
+			s.metrics.dirSyncFailures.Inc()
+			return 0, err
+		}
+		level.Error(s.logger).Log("msg", "loading block failed", "err", err, "blocks", failedBlocks)
```

Comment:
Does the logger support printing out string slice? I thought it is not supported and you need to convert it to string

Reply:
It was removed with the suggestion offline to use flag in `blockMetasFromOldest `